runtime/src/main/java/org/corfudb/runtime/collections/TxnContext.java

```diff
@@ -830,19 +831,21 @@ public Timestamp commit() {
         }
 
         // These can be moved to trace once stability improves.
-        log.debug("Txn committed on namespace {}", namespace);
-
-        // If we are here this means commit was successful, now invoke the callback with
-        // the final versions of all the mutated objects.
-        MultiObjectSMREntry writeSet = rootContext.getWriteSetInfo().getWriteSet();
-        final Map<String, List<CorfuStreamEntry>> mutations = new HashMap<>();
-        tablesInTxn.forEach((uuid, table) -> {
-            List<CorfuStreamEntry> writesInTable = writeSet.getSMRUpdates(uuid).stream().map(entry ->
-                    CorfuStreamEntry.fromSMREntry(entry, 0)).collect(Collectors.toList());
-            mutations.put(table.getFullyQualifiedTableName(), writesInTable);
-        });
-        commitCallbacks.forEach(cb -> cb.onCommit(mutations));
-        operations.clear();
+        log.trace("Txn committed on namespace {}", namespace);
+
+        if (txnType != READ_ONLY) {
```

Comment:
will it make a big difference?  any table which does not have writes gets skipped in the loop below.  So it is already a no-op.. right?

Reply:
the loop only iterates over the writeSet but it unconditionally invokes the callback for all registered commitCallbacks.
the callbacks have fixed costs overheads which are completely unnecessary for reads